.buildkite/scripts/publish/publish-common.sh

```diff
@@ -0,0 +1,31 @@
+#!/bin/bash
+
+if [[ "${CURDIR:-}" == "" ]]; then
+  echo "!! CURDIR is not set. Exiting."
+  exit 2
+fi
+
+function realpath {
+  echo "$(cd "$(dirname "$1")"; pwd)"/"$(basename "$1")";
+}
+
+export SCRIPT_DIR="$CURDIR"
+export BUILDKITE_DIR=$(realpath "$(dirname "$SCRIPT_DIR")")
+export PROJECT_ROOT=$(realpath "$(dirname "$BUILDKITE_DIR")")
+
+VERSION_PATH="$PROJECT_ROOT/product_version"
+export VERSION=$(cat $VERSION_PATH)
+
+if [[ "${USE_SNAPSHOT:-}" == "true" ]]; then
+  echo "Adding SNAPSHOT labeling"
+  export VERSION="${VERSION}-SNAPSHOT"
+fi
+
+export BASE_TAG_NAME=${DOCKER_IMAGE_NAME:-docker.elastic.co/enterprise-search/crawler}
+export DOCKERFILE_PATH=${DOCKERFILE_PATH:-Dockerfile}
+export DOCKER_ARTIFACT_KEY=${DOCKER_ARTIFACT_KEY:-crawler-docker}
+
+export VAULT_ADDR=${VAULT_ADDR:-https://secrets.elastic.co}
```

Comment:
I don't think we can read directly from this Vault within a Buildkite job without doing some extra work, _eg_ the 'transitive auth' approach I used here https://github.com/elastic/swiftype-ci-base/blob/master/.buildkite/hooks/pre-command

another option would be to duplicate the secrets in CI vault, in which case we could then use [this plugin](https://github.com/elastic/vault-secrets-buildkite-plugin) to make secret injection a bit nicer

Reply:
Unfortunately I don't have access to the example you've linked.
I think moving the credentials to CI vault makes sense. I'll do that (and look into using the plugin)

Reply:
I've moved it to the CI vault and updated everything.
I haven't tried to use the plugin. I can see it's cleaner but because everything is already written, I'd prefer not to rewrite it (also it's identical to Connectors which makes maintaining easier for the team).